appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -44,16 +45,42 @@ export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWiz
                     wizardContext.appInsightsComponent = await client.components.createOrUpdate(rgName, aiName, { kind: 'web', applicationType: 'web', location: appInsightsLocation });
                     const createdNewAppInsights: string = localize('createdNewAppInsights', 'Successfully created Application Insights resource "{0}".', aiName);
                     ext.outputChannel.appendLog(createdNewAppInsights);
+                } else if (pError.errorType === 'AuthorizationFailed') {
+                    if (!wizardContext.advancedCreation) {
+                        const createAppInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping Application Insights resource because you do not have permission to create one in this subscription.');
+                        ext.outputChannel.appendLog(createAppInsightsNotAvailable);
+                    } else {
+                        await this.selectExistingPrompt(wizardContext);
+                    }
                 } else {
                     throw error;
                 }
             }
         } else {
-            const appInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping creating an Application Insights resource because it isn\'t compatible with this location.');
+            const appInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping Application Insights resource because it isn\'t compatible with this location.');
             ext.outputChannel.appendLog(appInsightsNotAvailable);
         }
     }
 
+    public async selectExistingPrompt(wizardContext: IAppServiceWizardContext): Promise<void> {
+        const message: string = localize('aiForbidden', 'You do not have permission to create an app insights resource in subscription "{0}".', wizardContext.subscriptionDisplayName);
+        const selectExisting: MessageItem = { title: localize('selectExisting', 'Select Existing') };
+        const skipForNow: MessageItem = { title: localize('skipForNow', 'Skip for Now') };
+        wizardContext.telemetry.properties.cancelStep = 'AppInsightsNoPermissions';
+        const result = await wizardContext.ui.showWarningMessage(message, { modal: true }, selectExisting, skipForNow);
+
+        if (result === skipForNow) {
+            wizardContext.telemetry.properties.aiSkipForNow = 'true';
+            wizardContext.appInsightsSkip = true;
+        } else {
+            wizardContext.telemetry.properties.cancelStep = undefined;
```

Comment:
Move this cancelStep line just before `if (result === skipForNow) {` so that we also clear it for the skipForNow case